dom.go

```diff
@@ -454,6 +489,24 @@ func (tl *TokenList) Set(s []string) {
 	tl.SetString(strings.Join(s, " "))
 }
 
+type attribute struct {
+	key   string
+	value string
+}
+
+func (a attribute) Value() string {
+	return a.value
+}
+
+func (a attribute) Key() string {
+	return a.key
+}
+
+type Attribute interface {
```

Comment:
1. Is there any need for the indirection via an interface here? Couldn't we just export the struct and use it directly?
2. Can we use a map[string]string instead of []Attribute? Do `attributes` and `dataset` have defined order (in which case we couldn't use a map)?